src/main/java/org/opensearch/sdk/ExtensionNamedWriteableRegistry.java

```diff
@@ -79,12 +80,13 @@ private List<NamedWriteableRegistry.Entry> getNamedWriteables() {
      * @param request  The OpenSearch request to handle.
      * @return A response with a list of writeable names and fully qualified category class names to register within OpenSearch
      */
+    @SuppressWarnings("unchecked")
     public NamedWriteableRegistryResponse handleNamedWriteableRegistryRequest(OpenSearchRequest request) {
         logger.info("Registering Named Writeable Registry Request recieved from OpenSearch.");
         // Iterate through Extensions's named writeables and add to extension entries
-        Map<String, Class> extensionEntries = new HashMap<>();
+        Map<String, Class<? extends NamedWriteable>> extensionEntries = new HashMap<>();
         for (NamedWriteableRegistry.Entry entry : this.namedWriteables) {
-            extensionEntries.put(entry.name, entry.categoryClass);
+            extensionEntries.put(entry.name, (Class<? extends NamedWriteable>) entry.categoryClass);
```

Comment:
Note to other reviewers:  I blinked when I saw this direct field access and cast instead of a getter, but that's apparently the design, and the field is just a `<?>` despite being set with a `<T extends NamedWriteable>` in the constructor.  So this, surprisingly, is correct. 👀

Reply:
@dbwiddis you mean the right way to add generic type is `(Class<T extends NamedWriteable>)` instead of `(Class<? extends NamedWriteable>)`?

Reply:
No, you have it fine for this use, the `T` is in the `NamedWriteableRegistry.Entry` definition.